RecoMuon/MuonIdentification/interface/MuonShowerDigiFiller.h

```diff
@@ -46,6 +46,7 @@
 
 class MuonShowerDigiFiller {
 public:
+  MuonShowerDigiFiller() {}
```

Comment:
is this really needed with an empty body? it should already be the same as the implicit default.
please drop.

Reply:
Is there an implicit _default_ constructor? If I remove this line, I get a compilation error at the instantiation without parameters [here](https://github.com/trocino/cmssw/blob/fix_UBSAN_issue35036_MUO-MuIdProd/RecoMuon/MuonIdentification/plugins/MuonIdProducer.cc#L93). From the error message, it looks like there is an implicit copy constructor instead:

`/.../unique_ptr.h: In instantiation of 'typename std::_MakeUniq<_Tp>::__single_object std::make_unique(_Args&& ...) [with _Tp = MuonShowerDigiFiller; _Args = {}; typename std::_MakeUniq<_Tp>::__single_object = std::unique_ptr<MuonShowerDigiFiller>]':`

`/.../plugins/MuonIdProducer.cc:93:67:   required from here`
`/.../unique_ptr.h:962:30: error: no matching function for call to 'MuonShowerDigiFiller::MuonShowerDigiFiller()'`
`  962 |     { return unique_ptr<_Tp>(new _Tp(std::forward<_Args>(__args)...)); }`

`In file included from /.../plugins/MuonIdProducer.h:49,`
`                 from /.../plugins/MuonIdProducer.cc:12:`
`/.../interface/MuonShowerDigiFiller.h:49:3: note: candidate: 'MuonShowerDigiFiller::MuonShowerDigiFiller(const edm::ParameterSet&, edm::ConsumesCollector&&)'`
`   49 |   MuonShowerDigiFiller(const edm::ParameterSet&, edm::ConsumesCollector&& iC);`
`/.../interface/MuonShowerDigiFiller.h:49:3: note:   candidate expects 2 arguments, 0 provided`

`/.../interface/MuonShowerDigiFiller.h:47:7: note: candidate: 'MuonShowerDigiFiller::MuonShowerDigiFiller(const MuonShowerDigiFiller&)'`
`   47 | class MuonShowerDigiFiller {`
`/.../interface/MuonShowerDigiFiller.h:47:7: note:   candidate expects 1 argument, 0 provided`

`/.../interface/MuonShowerDigiFiller.h:47:7: note: candidate:` `'MuonShowerDigiFiller::MuonShowerDigiFiller(MuonShowerDigiFiller&&)'`
`/.../interface/MuonShowerDigiFiller.h:47:7: note:   candidate expects 1 argument, 0 provided`